File: bindings/java-ffm/src/main/java/org/maplibre/nativeffi/geo/ScreenBox.java
package org.maplibre.nativeffi.geo;

import java.util.Objects;

/** Screen-space box in logical map pixels. */
public record ScreenBox(ScreenPoint min, ScreenPoint max) {
  public ScreenBox {
    Objects.requireNonNull(min, "min");
    Objects.requireNonNull(max, "max");
  }
}

File: bindings/kotlin/src/commonMain/kotlin/org/maplibre/nativeffi/geo/ScreenBox.kt
package org.maplibre.nativeffi.geo

/**
 * Screen-space box in logical map pixels.
 *
 * Corners may be given in any order, and may extend past the viewport. Rendered queries normalize
 * the corners and clip the box to the viewport, so a box that over-covers the viewport queries
 * everything visible.
 */
public data class ScreenBox(public val min: ScreenPoint, public val max: ScreenPoint)
